app/screens/settings-screen/account/multi-account/profile.tsx

```diff
@@ -0,0 +1,216 @@
+import * as React from "react"
+import { ActivityIndicator, TouchableOpacity, View } from "react-native"
+
+import { ListItem, makeStyles, Overlay, useTheme, Text } from "@rn-vui/themed"
+import Icon from "react-native-vector-icons/Ionicons"
+import { useI18nContext } from "@app/i18n/i18n-react"
+import { useNavigation } from "@react-navigation/native"
+import { StackNavigationProp } from "@react-navigation/stack"
+import { RootStackParamList } from "@app/navigation/stack-param-lists"
+import { useState } from "react"
+import { useAppConfig } from "@app/hooks"
+import { testProps } from "@app/utils/testProps"
+import useLogout from "@app/hooks/use-logout"
+import { GaloyIconButton } from "@app/components/atomic/galoy-icon-button/galoy-icon-button"
+import Modal from "react-native-modal"
+import { GaloyPrimaryButton } from "@app/components/atomic/galoy-primary-button"
+import { GaloySecondaryButton } from "@app/components/atomic/galoy-secondary-button"
+import { toastShow } from "@app/utils/toast"
+
+export const ProfileScreen: React.FC<ProfileProps> = ({
+  identifier,
+  token,
+  nextProfileToken,
+  selected,
+  isFirstItem,
+  hasUsername,
+  lnAddressHostname,
+}) => {
+  const {
+    theme: { colors },
+  } = useTheme()
+  const styles = useStyles()
+  const { LL } = useI18nContext()
+
+  const navigation = useNavigation<StackNavigationProp<RootStackParamList>>()
+
+  const [switchLoading, setSwitchLoading] = useState(false)
+  const [logoutLoading, setLogoutLoading] = useState(false)
+  const [modalVisible, setModalVisible] = useState(false)
+
+  const { saveToken } = useAppConfig()
+  const { logout } = useLogout()
+
+  const handleProfileSwitch = async (nextToken?: string) => {
+    setSwitchLoading(true)
+    await saveToken(nextToken ?? token)
+
+    // Small delay to ensure the new session token is processed and updated in the global state before navigating.
+    // This prevents the "Primary" screen from initially loading data using the old session token.
+    await new Promise<void>((resolve) => {
+      setTimeout(() => resolve(), 100)
+    })
```

Comment:
we cant use timeouts every time we have a sync issue, please find the root cause